pandas/core/series.py

```diff
@@ -1585,6 +1585,56 @@ def __repr__(self) -> str:
 
         return result
 
+    def _repr_html_(self):
+        # TODO: Full independent HTML generation in SeriesFormatter, rather
+        # than depending on a limited subset of functionality via to_frame().
+
+        if get_option("display.notebook_repr_html"):
```

Comment:
As a follow up I think should put this in the NDFrame superclass shared by DataFrame and Series rather than copy paste